### PR TITLE
Add syntax support for field-level claims.

### PR DESCRIPTION
**Pull request by csilvestrini**
No DFA support, no proto encoding, just manifest parsing. Only for claims, not checks.

Let me revise my earlier estimate. I think this will take several days, not just 1 or 2.

### CODE REVIEW

**Comment by bgogul** [comment]
Thanks, Cameron. I will take a look at this PR shortly.

> Let me revise my earlier estimate. I think this will take several days, not just 1 or 2.

Given the revised estimate, is it still worth doing now?  I believe most of the work (except syntax) will carry over when we implement the full-blown proposal for field-level checks and claims. WDYT?

We could implement the full blown proposal directly, but that will take even more time.

File: src/runtime/manifest-ast-nodes.ts [hunk 1]
```diff
@@ -243,6 +243,7 @@ export interface Particle extends BaseNode {
 export interface ParticleClaimStatement extends BaseNode {
   kind: 'particle-trust-claim';
   handle: string;
+  fieldPath: string[];
```
Comment (bgogul) [inline src/runtime/manifest-ast-nodes.ts:246]: Don't we need to support other kinds of selectors, such as accessing an element in a collection, dereferencing, etc.?

**Comment by csilvestrini** [comment]
> Given the revised estimate, is it still worth doing now? I believe most of the work (except syntax) will carry over when we implement the full-blown proposal for field-level checks and claims. WDYT?

Yes I think it's still worth doing. It's not actually the syntax that is the slow bit, it's updating the data structures and subsequent processing to support the new concept. We'll have to do that regardless of what the syntax looks like (and I don't expect it will need to change much if at all if we change the syntax).

It'll take longer than I thought just because there are so many different variations to update: claims in particles, claims on stores, plus a bunch of checks variations I haven't investigated yet.

File: src/runtime/manifest-ast-nodes.ts (hunk 1 above)
Comment (csilvestrini) [reply src/runtime/manifest-ast-nodes.ts:246]: My intention was to use dots for all of those selectors too. e.g. `someFieldCollection.someReference.someOtherField` would descend into the collection's inner type, and into the reference's inner type too. 

I'll add appropriate tests for all those cases once I add the code to validate the fields.

Comment (bgogul) [reply src/runtime/manifest-ast-nodes.ts:246]: How would we know what the kind of selector is? Do we infer it from the type essentially?

**Pull request merged**